index.js

```diff
@@ -65,70 +63,355 @@ export class MentionsTextInput extends Component {
+  handleClick(position) {
+    if (!this.triggerMatrix || !this.triggerMatrix.length) {
+      return;
+    }
+
+    const index = this.getSubsequentTriggerIndex(position);
+    this.handleDeleteTriggerOnBackspace(position, index);
+
+    if (this.isPositionWithinTrigger(position, index)) {
+      this.startTracking(position, index);
+      return;
+    }
+
+    if (position === -1 || this.state.text && this.state.text[position] === ' ') {
+      this.stopTracking();
+    }
   }
 
-  updateSuggestions(lastKeyword) {
-    this.props.triggerCallback(lastKeyword);
+  handleTriggerSplitBySpace(position) {
+    if (!this.triggerMatrix
+          || !this.triggerMatrix.length
+          || !this.isTrackingStarted
+          || position < 1
+          || position >= this.state.text.length) {
+      return;
+    }
+
+    const index = this.getSubsequentTriggerIndex(position);
+    this.triggerMatrix[index][1] = position - 1;
   }
 
-  identifyKeyword(val) {
-    if (this.isTrackingStarted) {
-      const boundary = this.props.triggerLocation === 'new-word-only' ? 'B': '';
-      const delay = this.props.triggerDelay ? this.props.triggerDelay : 0;
-      const pattern = new RegExp(`\\${boundary}${this.props.trigger}[a-z0-9_-]{${delay},}`, `gi`);
-      const keywordArray = val.match(pattern);
-      if (keywordArray && !!keywordArray.length) {
-        const lastKeyword = keywordArray[keywordArray.length - 1];
-        this.updateSuggestions(lastKeyword);
-        this.openSuggestionsPanel();
-      } else {
-        this.closeSuggestionsPanel();
+  isTriggerSplitBySpace(position) {
+    if (!this.triggerMatrix || !this.triggerMatrix.length) {
+      return false;
+    }
+
+    const index = this.getSubsequentTriggerIndex(position);
+    return this.isPositionWithinTrigger(position, index)
+              && this.isTrackingStarted
+              && this.state.text[position] === ' ';
+  }
+
+  getDistanceToNextSpace(index = -1) {
+    if (index === -1 || !this.state.text || !this.state.text.length || index > this.state.text.length) {
+      return 0;
+    }
+
+    const spaceIndex = this.state.text.indexOf(' ', index);
+    return spaceIndex === -1 ? this.state.text.length - 1 - index : spaceIndex - index;
+  }
+
+  getTriggerKeyword(position, index = -2) {
+    if (!this.triggerMatrix || !this.triggerMatrix.length || !this.isTrackingStarted) {
+      return;
+    }
+
+    if (index === -2) {
+      index = this.getSubsequentTriggerIndex(position);
+    }
+
+    if (index === -1 || index >= this.triggerMatrix.length) {
+      return;
+    }
+
+    const start = this.triggerMatrix[index][0];
+    const end = this.triggerMatrix[index][1];
+    const pattern = new RegExp(`${this.props.trigger}[a-z0-9_-]*`, `gi`);
+    const triggerText = this.state.text.slice(start, end + this.getDistanceToNextSpace(end) + 1);
+    const keywordArray = triggerText.match(pattern);
+
+    return keywordArray && !!keywordArray.length ? keywordArray[0] : '';
```

Comment:
Why `!!` vs just `keywordArray.length`?

Reply:
its the strict true false evaluation of the length. just let it like that from the original repo. going to change it

Reply:
I'm fine with that then.